src/app/kademlia-haskell/release2.nix

```diff
@@ -1,26 +1,8 @@
-{ compiler ? "ghc863" }:
-
-let
-  config = {
-    packageOverrides = pkgs: rec {
-      haskell = pkgs.haskell // {
-        packages = pkgs.haskell.packages // {
-          "${compiler}" = pkgs.haskell.packages."${compiler}".override {
-            overrides = haskellPackagesNew: haskellPackagesOld: rec {
-              kademlia = haskellPackagesNew.callPackage ./prefetch/kademlia1101.nix { };
-
-              project1 =
-                haskellPackagesNew.callPackage ./default.nix { };
-            };
-          };
-        };
-      };
-    };
-  };
-
-  pkgs = import <nixpkgs> { inherit config; };
-
-in
-  { project1 = pkgs.haskell.packages.${compiler}.project1;
-  }
-
+# We use stack.yaml as a single source of truth for our Haskell builds. You can
```

Comment:
Not clear what "source of truth" means. What information is contained in that file? Where else might conflicting information come from?

Reply:
Source of truth as in the definition of what our Haskell dependencies are and how to build the executable. The old method you could build it using Stack, and it would get that information from `stack.yaml`, or you could build it using Nix and it would get that information from the various `.nix` files. The two configurations could differ, since they were independent. In this method the Nix expressions are determined by the content of the `stack.yaml` file.